mmdet/core/evaluation/eval_hooks.py

```diff
@@ -1,14 +1,47 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+import bisect
 import os.path as osp
 
+import mmcv
 import torch.distributed as dist
-from mmcv.runner import DistEvalHook as BaseDistEvalHook
 from mmcv.runner import EvalHook as BaseEvalHook
 from torch.nn.modules.batchnorm import _BatchNorm
 
 
 class EvalHook(BaseEvalHook):
 
+    def __init__(self, *args, dynamic_intervals=None, **kwargs):
+        super(EvalHook, self).__init__(*args, **kwargs)
+
+        self.dynamic_intervals = dynamic_intervals
+        if dynamic_intervals is not None:
+            assert mmcv.is_list_of(dynamic_intervals, tuple)
+
+            self.dynamic_steps = [0]
+            self.dynamic_steps.extend([
+                dynamic_interval[0] for dynamic_interval in dynamic_intervals
+            ])
+            self.dynamic_values = [self.interval]
+            self.dynamic_values.extend([
+                dynamic_interval[1] for dynamic_interval in dynamic_intervals
+            ])
```

Comment:
The name `dynamic_steps` and `dynamic_values` are not clear enough.
```suggestion
        self.use_dynamic_intervals = dynamic_intervals is not None
        if self.use_dynamic_intervals:
            assert mmcv.is_list_of(dynamic_intervals, tuple)

            self.dynamic_milestones = [0]
            self.dynamic_milestones.extend([
                dynamic_interval[0] for dynamic_interval in dynamic_intervals
            ])
            self.dynamic_intervals = [self.interval]
            self.dynamic_intervals.extend([
                dynamic_interval[1] for dynamic_interval in dynamic_intervals
            ])
```